config/config.go

```diff
@@ -174,54 +107,52 @@ type GrafanaDashboardConfig struct {
 }
 
 type GrafanaVariablesConfig struct {
-	Namespace string `yaml:"namespace" json:"namespace,omitempty"`
 	App       string `yaml:"app" json:"app,omitempty"`
+	Namespace string `yaml:"namespace" json:"namespace,omitempty"`
 	Service   string `yaml:"service" json:"service,omitempty"`
 	Version   string `yaml:"version" json:"version,omitempty"`
 	Workload  string `yaml:"workload" json:"workload,omitempty"`
 }
 
 // TracingConfig describes configuration used for tracing links
 type TracingConfig struct {
-	// Enable autodiscover and Jaeger in Kiali
-	Enabled      bool   `yaml:"enabled"`
+	Auth         Auth   `yaml:"auth"`
+	Enabled      bool   `yaml:"enabled"` // Enable autodiscover and Jaeger in Kiali
+	InClusterURL string `yaml:"in_cluster_url"`
 	Namespace    string `yaml:"namespace"`
-	Service      string `yaml:"service"`
+	Path         string `yaml:"-"` // Path stores the value of QUERY_BASE_PATH, not configurable in ConfigMap
 	Port         int32  `yaml:"port"`
+	Service      string `yaml:"service"`
 	URL          string `yaml:"url"`
-	Auth         Auth   `yaml:"auth"`
-	InClusterURL string `yaml:"in_cluster_url"`
-	// Path store the value of QUERY_BASE_PATH
-	Path string `yaml:"-"`
 }
 
 // IstioConfig describes configuration used for istio links
 type IstioConfig struct {
-	UrlServiceVersion      string `yaml:"url_service_version"`
 	IstioIdentityDomain    string `yaml:"istio_identity_domain,omitempty"`
 	IstioSidecarAnnotation string `yaml:"istio_sidecar_annotation,omitempty"`
+	UrlServiceVersion      string `yaml:"url_service_version"`
 }
 
 // ThreeScaleConfig describes configuration used for 3Scale adapter
 type ThreeScaleConfig struct {
 	AdapterName    string `yaml:"adapter_name"`
-	AdapterService string `yaml:"adapter_service"`
 	AdapterPort    string `yaml:"adapter_port"`
+	AdapterService string `yaml:"adapter_service"`
 }
 
 // ExternalServices holds configurations for other systems that Kiali depends on
 type ExternalServices struct {
+	Grafana    GrafanaConfig    `yaml:"grafana,omitempty"`
 	Istio      IstioConfig      `yaml:"istio,omitempty"`
 	Prometheus PrometheusConfig `yaml:"prometheus,omitempty"`
-	Grafana    GrafanaConfig    `yaml:"grafana,omitempty"`
-	Tracing    TracingConfig    `yaml:"tracing,omitempty"`
 	ThreeScale ThreeScaleConfig `yaml:"threescale,omitempty"`
+	Tracing    TracingConfig    `yaml:"tracing,omitempty"`
 }
 
 // LoginToken holds config used in token-based authentication
 type LoginToken struct {
-	SigningKey        string `yaml:"signing_key,omitempty"`
 	ExpirationSeconds int64  `yaml:"expiration_seconds,omitempty"`
+	SigningKey        string `yaml:"signing_key,omitempty"`
```

Comment:
Should SigningKey be treated as a potential overwritten property ? As a token or password ?

Reply:
Yeah, that's a good one. We can add that. Let me add another commit for that.